openapi/endpoints/funding-sources/models/funding-source-interaction.yaml

```diff
@@ -0,0 +1,77 @@
+type: object
+properties:
+  id:
+    type: string
+    description: id of the Funding Source Interaction.
+    example: f0220f084a182e3f4b4d605cda1d3340
+  fundingSourceId:
+    type: string
+    description: Funding Source to which the interaction belongs to.
+    example: 2be03cd65da12fe29b8910af7b4b85e7
+  accountId:
+    type: string
+    description: The accountId to which the interaction belongs to.
+    example: 979254377d0e05002242d038926a5691
+  type:
+    type: string
+    description: The interaction type.
+    enum: 
+      - Deposit
+      - Withdrawal
+      - Refund
+      - Payment
+    example: Payment
+  status:
+    type: string
+    description: The current status of the interaction. It will vary depending on the interaction type.
+    example: processing-successful
+  amount:
+    type: number
+    description: The minor units amount of the transaction.
+    example: 20000
+  token:
+    type: string
+    description: The token used in the transaction.
+    example: USDC
+  createdAt:
+    type: string
+    description: Date time on which the interaction was created.
+    example: '2023-11-15T00:48:48.902Z'
+  modifiedAt:
+    type: string
+    description: Last modified date of this interaction.
+    example: '2023-11-15T00:49:59.261Z'
+  context:
+    type: object
+    properties:
+      ref:
+        type: string
+        description: A context reference based on the type of the interaction. It could be a `transactionId` in the case of a payment, or an `eventId` in the case of an smart contract event.
+        example: '1000000488078'
+      type:
+        type: string
+        description: A context type for the context reference, which will depend on the interaction type. 
```

Comment:
this doesnt sound right. any better suggestions on the description for `ref` ? @Null000 @javcode